routers/order.js

```diff
@@ -0,0 +1,94 @@
+const { Router } = require("express");
+const { mealTopping, mealComposition, order, topping, salsa, meal, filling } = require("../models");
+
+const router = new Router();
+let clients = [];
+
+router.post('/', async (req,res,next) => {
+  
+  const { customerAddress, customerPhone, mealCompositions, totalPrice, comment } = req.body.order;
+
+  if (!customerPhone || !mealCompositions.length > 0) {
+    res.status(400).send({ message: 'Order not complete' });
+  }
+	try{
+    
+    const newOrder = await order.create({
+      customerAddress,
+      customerPhone,
+      totalPrice,
+      comment
+    });
+    const { id: orderId } = newOrder;
+
+    await Promise.all(mealCompositions.map(async (meal) => {
+      const { meal: mealId, filling: fillingId, salsa: salsaId, toppings } = meal;
+      const newMealComposition = await mealComposition.create({
+        orderId,
+        mealId,
+        fillingId,
+        salsaId
+      });      
+      const { id: mealCompositionId } = newMealComposition;
+      await Promise.all(toppings.map(async (topping) => {
+        const newMealTopping = await mealTopping.create({
+          toppingId: topping,
+          mealCompositionId
+        });
+        return newMealTopping;
+      }));
+      return newMealComposition;
+    }));
```

Comment:
nice bit of logic. I would advice you to use `bulkCreate` for this kind of situation as its more appropiate than doing one `create` per iteration of the `.map`.
Also, it removes the need for `Promise.all`. An example with this inner map:
```
const { id: mealCompositionId } = newMealComposition;

// we create the array of objects to insert
const mealToppingsToCreate = toppings.map(t => ({ toppingId: t, mealCompositionId }) 

const created = await mealTopping.bulkCreate(mealToppingsToCreate);

```